src/components/Navbar/navbar.js

```diff
@@ -0,0 +1,131 @@
+import React, { useState } from 'react';
+import { NavLink } from 'react-router-dom';
+import {
+  FaFacebookF, FaGooglePlusG, FaVimeoV, FaPinterestP,
+} from 'react-icons/fa';
+import { BsTwitter } from 'react-icons/bs';
+import { AiOutlineCopyrightCircle } from 'react-icons/ai';
+import { GiHamburgerMenu } from 'react-icons/gi';
+import { toast } from 'react-toastify';
+import logoImage from '../../Assets/logolibrary.png';
+import '../../stylesheets/navbar.css';
+import Logout from '../user/logout';
+
+const SideNav = () => {
+  const [isNavVisible, setNavVisible] = useState(false);
+  const toggleSidebar = () => {
+    setNavVisible(!isNavVisible);
+  };
+
+  const [isLogoutModalOpen, setLogoutModalOpen] = useState(false);
+
+  const onLogout = () => {
+    setLogoutModalOpen(true);
+  };
+
+  const closeLogoutModal = () => {
+    setLogoutModalOpen(false);
+  };
+
+  const userData = JSON.parse(localStorage.getItem('user'));
+  const { role } = userData;
+
+  const adminWarn = () => {
+    if (role !== 'admin') {
+      toast.error('You are not an admin');
+    }
+  };
+
+  return (
+    <div className={`nav-container ${isNavVisible ? 'open' : ''} d-flex flex-column align-items-center`}>
+
+      <nav className="burger">
+        <button type="button" aria-label="Navigation" className="toggle-button" onClick={toggleSidebar}>
+          <GiHamburgerMenu />
+        </button>
+      </nav>
+      <nav className={`nav ${isNavVisible ? 'show' : 'hidden'}`}>
+        <div className="logo-img">
+          <img className="logo" src={logoImage} alt="logo" />
+        </div>
+        <ul className="sidenav-list d-flex flex-column justify-content-center">
+          <li className="sidenav-item">
+            <NavLink to="/MainPage" className="sidenav-link" onClick={toggleSidebar}>
+              Books
+            </NavLink>
+          </li>
+          { role === 'admin' && (
+          <>
+            <li className="sidenav-item">
+              <NavLink to="/update" className="sidenav-link" onClick={toggleSidebar}>
+                Give Permission
+              </NavLink>
+            </li>
+          </>
+          )}
+          <li className="sidenav-item">
+            <NavLink to="/BookForm" className="sidenav-link" onClick={() => { toggleSidebar(); adminWarn(); }}>
+              Add Book +
+            </NavLink>
+          </li>
+          <li className="sidenav-item">
+            <NavLink to="/deletebook" className="sidenav-link" onClick={() => { toggleSidebar(); adminWarn(); }}>
+              Delete Book -
+            </NavLink>
+          </li>
```

Comment:
- Thanks for your comment regarding this point. I observed the requirement too, now, a better implementation for this could be to hide the options for no-admin users in place to only show an alert for no-admin. What do you think about this? 🐶 

* This is the only required change here, after you hide this, please open another code review request and this will be done. Thanks for your willingness to continue to work on it.

Reply:
Hello, thank you for replying back and continuing on providing feedback.

We do agree of that change and we could look into adding it. We are still weighing on if we wish to go with a different route than the project requirements. The team and I will try to come up with a compromise for the option (since we did something similar with 'give permission' already)